### PR TITLE
More bcf fixes

### PR DESCRIPTION
**Pull request by bioinformed**
* Issue #371: do not interpret vector end codes for scalar values	
* Issue #355: Implement streams using file descriptors for VariantFile
* Fix Python 3 build on Darwin
* Refactor common functionality from AlignmentFile and VariantFile into a common HTSFile base class

### CODE REVIEW

**Comment by bioinformed** [comment]
@AndreasHeger: You may want to review this PR, since it adds a common base class `HTSFile` for `AlignmentFile` and  `VariantFile` and exposes more `htsFormat` information.

**Comment by kyleabeauchamp** [comment]
Should this base class be used for TabixFile as well?  It seems to have at least some of the same fields (e.g. filename, index).

**Comment by bioinformed** [comment]
Good point, @kyleabeauchamp!   `TabixFile` now inherits from `HTSFile` too.

File: pysam/libctabix.pyx [hunk 1]
```diff
@@ -72,7 +72,7 @@ from pysam.libchtslib cimport htsFile, hts_open, hts_close, HTS_IDX_START,\
     BGZF, bgzf_open, bgzf_dopen, bgzf_close, bgzf_write, \
     tbx_index_build, tbx_index_load, tbx_itr_queryi, tbx_itr_querys, \
     tbx_conf_t, tbx_seqnames, tbx_itr_next, tbx_itr_destroy, \
-    tbx_destroy, hisremote
+    tbx_destroy, hisremote, region_list
```
Comment (kyleabeauchamp) [inline pysam/libctabix.pyx]: What's the deal with `region_list`?  It's newly imported here but commented out later.

**Comment by AndreasHeger** [comment]
Thanks, having HTSFile is clearly the right thing to do.

**Pull request merged**